### ISSUE TITLE
Introduce streaming-data-loader to address memory pressure on model training with Huge training dataset.

### ISSUE THREAD

**Issue by tybalex** [rancher/opni#1095]
the current implementation of data loader of Opni log anomaly detection's deep learning model loads all the training dataset into memory. When the training dataset is huge, such approach will put high pressure on memory or even blow out the nodes. 
This task aims to introduce a streaming way to load the data use either generator or a streaming dataloader, which will only keep a small amount of data in memory once a time and solve the issue.

EPIC: https://github.com/rancher/opni/issues/1094

PRs:
* https://github.com/rancher/opni-inference-service/pull/40. This includes the implementation of the streaming dataloader, IterablePaddedDataset and other changes in the opni-inference-service repo. 
* https://github.com/rancher/opni-training-controller/pull/17. Minor changes to adjust to the introduction of streaming dataloader

**Comment by tybalex** [comment]
The testing actually takes longer. There are a few minor issue found within this design:
1. unstable model quality -- with the streaming dataloader, data can't get shuffled , so some normal data might not get trained well and be marked as Anomaly
2. Opensearch scroll timeout -- masking is slow and training is not fast, which results in the OS query scroll timeout
3. when test with `sock-shop`, the vocabulary increase super fast. Maybe need to filter vocab that appears just once.

**Comment by tybalex** [comment]
Will further improve the feature and address above issues in Sprint 3. Close issue
